File: tests/dynamic_fixtures/python/ssrf_positive.py
"""SSRF — positive fixture.

Vulnerable function: fetches a user-controlled URL.
Expected verdict: Confirmed (file:// payload reads /etc/passwd → "root:").
"""
import urllib.request


def fetch_url(url):
    """Vulnerable: fetches URL provided by user without validation."""
    try:
        with urllib.request.urlopen(url, timeout=3) as resp:
            content = resp.read().decode("utf-8", errors="replace")
            print(content)
    except Exception as e:
        print(f"Fetch error: {e}", end="")

File: tests/dynamic_fixtures/rust/ssrf_positive.rs
/// SSRF — positive fixture.
///
/// Vulnerable function: fetches a user-controlled URL. Implements a minimal
/// file:// scheme reader so the test requires no network and no async runtime.
///
/// Expected verdict: Confirmed (payload "file:///etc/passwd" causes "daemon:"
/// to appear in stdout via the file:// scheme handler).
/// Cap: SSRF  Entry: `run(payload: &str)`
pub fn run(payload: &str) {
    println!("__NYX_SINK_HIT__");
    let _ = std::io::Write::flush(&mut std::io::stdout());

    // Vulnerable: user controls the URL — SSRF via file:// scheme reaches local files.
    let result = fetch_url(payload);
    print!("{}", result);
}

fn fetch_url(url: &str) -> String {
    if let Some(path) = url.strip_prefix("file://") {
        std::fs::read_to_string(path)
            .unwrap_or_else(|e| format!("fetch error: {}", e))
    } else {
        // For non-file schemes, report the target (demonstrating SSRF intent).
        format!("SSRF: would connect to {}", url)
    }
}
